app/src/main/java/io/pslab/activity/PowerSourceActivity.java

```diff
@@ -248,26 +240,11 @@ public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
             @Override
             public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
                 if (actionId == EditorInfo.IME_ACTION_DONE) {
-                    String voltageValue = displayPV3.getText().toString();
-                    voltageValue = voltageValue.replace("V", "");
-                    if (NumberUtils.isCreatable(voltageValue)) {
-                        float voltage = Float.parseFloat(voltageValue);
-                        if (voltage < 0.00f) {
-                            voltage = 0.00f;
-                            displayPV3.setText(voltage + " V");
-                        }
-                        if (voltage > 3.30f) {
-                            voltage = 3.30f;
-                            displayPV3.setText(voltage + " V");
-                        }
-                        controllerPV3.setProgress(mapPowerToProgress(voltage, PV3_CONTROLLER_MAX,
-                                3.30f, 0.00f));
-                    } else {
-                        float voltage = 0.00f;
-                        displayPV3.setText(voltage + " V");
-                        controllerPV3.setProgress(mapPowerToProgress(voltage, PV3_CONTROLLER_MAX,
-                                3.30f, 0.00f));
-                    }
+                    final String voltageValue = remove(displayPV3.getText(), "V", "\\+").trim();
```

Comment:
So, currently we are not supporting negative numbers to be input via the _EditTexts_ (we have `android:inputType` set to `numberDecimal` only for _displayPV1_ and _displayPV2_). 
However, those controllers can support negative voltages as well, and we should implement negative inputs. But when we do implement those, we'd be having many corner cases to handle so as to make sure that the user input is float-parseable. Wouldn't that increase the complexity of the operations we perform on the string before parsing it ? Or is it handleable using this approach ?
Please let me know if this concern sounds right.

Reply:
I tried setting the `android:inputType` to `numberSigned|numberDecimal` which nearly did the trick, but unfortunately the keyboard of my test phone is more or less broken. The keyboard has point and comma, but the comma is disabled, even when using the German locale. I can only use the point, but then the String cannot be parsed.

There are more or less clunky workarounds for this problem (see https://stackoverflow.com/questions/3821539/decimal-separator-comma-with-numberdecimal-inputtype-in-edittext/51983392), but I would propose that we keep the current behavior for now and I will create an extra issue for this problem.

Reply:
@marcnause Sure!